plugins/command-line/prism-command-line.js

```diff
@@ -4,33 +4,99 @@ if (typeof self === 'undefined' || !self.Prism || !self.document) {
 	return;
 }
 
-Prism.hooks.add('complete', function (env) {
-	if (!env.code) {
+Prism.hooks.add('before-highlight', function (env) {
+	env.vars = env.vars || {};
+	env.vars['command-line'] = env.vars['command-line'] || {};
+
+	if (env.vars['command-line'].complete || !env.code) {
+		env.vars['command-line'].complete = true;
 		return;
 	}
 
 	// Works only for <code> wrapped inside <pre> (not inline).
 	var pre = env.element.parentNode;
 	var clsReg = /\s*\bcommand-line\b\s*/;
-	if (
-		!pre || !/pre/i.test(pre.nodeName) ||
-			// Abort only if neither the <pre> nor the <code> have the class
-		(!clsReg.test(pre.className) && !clsReg.test(env.element.className))
-	) {
+	if (!pre || !/pre/i.test(pre.nodeName) || // Abort only if neither the <pre> nor the <code> have the class
+		(!clsReg.test(pre.className) && !clsReg.test(env.element.className))) {
+		env.vars['command-line'].complete = true;
 		return;
 	}
 
-	if (env.element.querySelector('.command-line-prompt')) {
-		// Abort if prompt already exists.
+	if (env.element.querySelector('.command-line-prompt')) { // Abort if prompt already exists.
+		env.vars['command-line'].complete = true;
 		return;
 	}
 
-	if (clsReg.test(env.element.className)) {
-		// Remove the class "command-line" from the <code>
+	var codeLines = env.code.split('\n');
+	env.vars['command-line'].numberOfLines = codeLines.length;
+	env.vars['command-line'].outputLines = [];
+
+	var outputSections = pre.getAttribute('data-output');
+	var outputFilter = pre.getAttribute('data-filter-output');
+	if (outputSections || outputSections === '') { // The user specified the output lines. -- cwells
+		outputSections = outputSections.split(',');
+		for (var i in outputSections) { // Parse the output sections into start/end ranges. -- cwells
+			var range = outputSections[i].split('-');
+			var outputStart = parseInt(range[0]);
```

Comment:
You might want to specify `parseInt`'s second parameter to `10`, just to make `012-013` work the same as `12-13`.